libs/isograph-react/src/core/cache.ts

```diff
@@ -335,15 +352,29 @@ function callSubscriptions(
   );
 }
 
-function hasOverlappingIds(set1: Set<DataId>, set2: Set<DataId>): boolean {
-  for (const id of set1) {
-    if (set2.has(id)) {
-      return true;
+function hasOverlappingIds(
+  set1: EncounteredIds,
+  set2: EncounteredIds,
+): boolean {
+  for (const typeName in set1) {
+    const set2Ids = set2[typeName];
+    if (!set2Ids) {
+      continue;
+    }
+
+    for (const id of set1[typeName]) {
+      if (set2Ids.has(id)) {
+        return true;
+      }
     }
   }
   return false;
 }
 
+export interface EncounteredIds {
```

Comment:
any reason this is an interface instead of a type, and any reason this isn't `Map<TypeName, Set<DataId>>`?